src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -2049,6 +2049,33 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         //   the \n (but not including it). Then, we check the if the
         //   previous character is \r, if its not, then we had a lone \n
         //   and so we append our own \r
+        // - For security reasons, most control characters should be removed.
+
+        auto pred = [](wchar_t c) {
+            if (c >= L'\x20' && c <= L'\x7f')
+            {
+                // Printable ASCII characters + DEL.
+                return false;
+            }
+
+            if (c > L'\x7f')
+            {
+                // Not a control character.
+                return false;
+            }
+
+            // All ASCII control characters will be removed except HT(0x09), LF(0x0a),
+            // CR(0x0d) and DEL(0x7F).

```

Comment:
Removing `CR` and `LF` seems meaningless from a practical point of view. But generally I think this function *should* remove `CR` and `LF` had it not been in this particular context.

Reply:
(copied from #7508)

The best resource I can find is here https://security.stackexchange.com/a/52655. It suggest removing every control characters except:

* CR(0x0d) \r 
* LF(0x0a) \n 
* HT(0x09) \t, 
* DEL(0x7F)

Then there's some other opinions in https://bugzilla.gnome.org/show_bug.cgi?id=753197, regarding removing:

* 0x80-0x9F

And here https://bugzilla.gnome.org/show_bug.cgi?id=794653, regarding removing:

* BS(0x08)
* DEL(0x7F)